librz/cons/dietline.c

```diff
@@ -297,121 +298,124 @@ static inline bool is_word_break_char(char ch, bool mode) {
-	undo_add_entry(i, rz_str_ndup(I.clipboard, len), NULL);
-	I.buffer.length -= len;
+	len = i - line->buffer.index;
+	free(line->clipboard);
+	line->clipboard = rz_str_ndup(line->buffer.data + line->buffer.index, len);
+	rz_line_clipboard_push(line, line->clipboard);
+	memmove(line->buffer.data + line->buffer.index, line->buffer.data + i, line->buffer.length - i + 1);
+	undo_add_entry(line, i, rz_str_ndup(line->clipboard, len), NULL);
+	line->buffer.length -= len;
 }
 
-static void paste(bool *enable_yank_pop) {
-	if (I.clipboard) {
-		char *cursor = I.buffer.data + I.buffer.index;
-		int dist = (I.buffer.data + I.buffer.length) - cursor;
-		int len = strlen(I.clipboard);
-		I.buffer.length += len;
-		memmove(cursor + len, cursor, dist);
-		memcpy(cursor, I.clipboard, len);
-		undo_add_entry(I.buffer.index, NULL, rz_str_ndup(I.clipboard, len));
-		I.buffer.index += len;
-		*enable_yank_pop = true;
+static void paste(RzLine *line, bool *enable_yank_pop) {
+	if (!line->clipboard) {
+		return;
 	}
+	char *cursor = line->buffer.data + line->buffer.index;
+	int dist = (line->buffer.data + line->buffer.length) - cursor;
+	int len = strlen(line->clipboard);
+	line->buffer.length += len;
+	memmove(cursor + len, cursor, dist);
+	memcpy(cursor, line->clipboard, len);
+	undo_add_entry(line, line->buffer.index, NULL, rz_str_ndup(line->clipboard, len));
+	line->buffer.index += len;
+	*enable_yank_pop = true;
 }
 
-static void unix_word_rubout(void) {
+static void unix_word_rubout(RzLine *line) {
 	int i, len;
-	if (I.buffer.index > 0) {
-		for (i = I.buffer.index - 1; i > 0 && I.buffer.data[i] == ' '; i--) {
-			/* Move cursor backwards until we hit a non-space character or EOL */
-			/* This removes any trailing spaces from the input */
-		}
-		for (; i > 0 && I.buffer.data[i] != ' '; i--) {
-			/* Move cursor backwards until we hit a space character or EOL */
-			/* This deletes everything back to the previous space character */
-		}
-		if (i > 0) {
-			i++;
-		} else if (i < 0) {
-			i = 0;
-		}
-		if (I.buffer.index > I.buffer.length) {
-			I.buffer.length = I.buffer.index;
-		}
-		len = I.buffer.index - i;
-		I.clipboard = rz_str_ndup(I.buffer.data + i, len);
-		rz_line_clipboard_push(I.clipboard);
-		undo_add_entry(i, rz_str_ndup(I.clipboard, len), NULL);
-		memmove(I.buffer.data + i,
-			I.buffer.data + I.buffer.index,
-			I.buffer.length - I.buffer.index + 1);
-		I.buffer.length = strlen(I.buffer.data);
-		I.buffer.index = i;
+	if (line->buffer.index < 1) {
+		return;
+	}
+	for (i = line->buffer.index - 1; i > 0 && line->buffer.data[i] == ' '; i--) {
+		/* Move cursor backwards until we hit a non-space character or EOL */
+		/* This removes any trailing spaces from the input */
+	}
+	for (; i > 0 && line->buffer.data[i] != ' '; i--) {
+		/* Move cursor backwards until we hit a space character or EOL */
+		/* This deletes everything back to the previous space character */
+	}
+	if (i > 0) {
+		i++;
+	} else if (i < 0) {
+		i = 0;
```

Comment:
this should never happen. check if coverity says is dead code

Reply:
Why? The previous two `for` cycles can make it negative.